src/main/java/frc/trigon/robot/subsystems/arm/simulationarm/SimulationArmIO.java

```diff
@@ -0,0 +1,110 @@
+    }
+
+    @Override
+    protected void setTargetAngleState(TrapezoidProfile.State targetState) {
+        setAngleMotorsPower(calculateAnglePowerFromState(targetState));
+    }
+
+    @Override
+    protected void setTargetElevatorState(TrapezoidProfile.State targetState) {
+        setElevatorMotorsPower(calculateElevatorPowerFromState(targetState));
+    }
+
+    @Override
+    protected void stopAngleMotors() {
+        setAngleMotorsPower(0);
+    }
+
+    @Override
+    protected void stopElevatorMotors() {
+        setElevatorMotorsPower(0);
+    }
+
+    private double calculateElevatorPowerFromState(TrapezoidProfile.State targetState) {
+        double pidOutput = SimulationArmConstants.ELEVATOR_PID_CONTROLLER.calculate(
+                getElevatorPositionRevolutions(),
+                targetState.position
+        );
+        double feedforward = SimulationArmConstants.ELEVATOR_FEEDFORWARD.calculate(targetState.velocity);
+        return pidOutput + feedforward;
+    }
+
+    private double calculateAnglePowerFromState(TrapezoidProfile.State targetState) {
+        double pidOutput = SimulationArmConstants.ANGLE_PID_CONTROLLER.calculate(
+                getAnglePositionDegrees(),
+                targetState.position
+        );
+        double feedforward = SimulationArmConstants.ANGLE_FEEDFORWARD.calculate(
+                Units.degreesToRadians(targetState.position),
+                targetState.velocity
+        );
+        return pidOutput + feedforward;
+    }
+
+    private double getElevatorPositionRevolutions() {
+        return Conversions.distanceToRevolutions(elevatorSimulation.getPositionMeters(), SimulationArmConstants.ELEVATOR_METERS_PER_REVOLUTION);
+    }
+
+    private double getElevatorVelocityRevolutionsPerSecond() {
+        return Conversions.distanceToRevolutions(elevatorSimulation.getVelocityMetersPerSecond(), SimulationArmConstants.ELEVATOR_METERS_PER_REVOLUTION);
+    }
+
+    private double getAnglePositionDegrees() {
+        return Units.radiansToDegrees(angleSimulation.getAngleRads());
+    }
+
+    private double getAngleVelocityDegreesPerSecond() {
+        return Units.radiansToDegrees(angleSimulation.getVelocityRadPerSec());
+    }
+
+    private void setAngleMotorsPower(double power) {
+        angleMotorVoltage = MathUtil.clamp(
+                powerToVoltage(power),
+                -SimulationArmConstants.VOLTAGE_COMPENSATION_SATURATION,
+                SimulationArmConstants.VOLTAGE_COMPENSATION_SATURATION
+        );
+        angleSimulation.setInputVoltage(power);
+    }
+
+    private void setElevatorMotorsPower(double power) {
+        elevatorMotorVoltage = MathUtil.clamp(
+                powerToVoltage(power),
+                -SimulationArmConstants.VOLTAGE_COMPENSATION_SATURATION,
+                SimulationArmConstants.VOLTAGE_COMPENSATION_SATURATION
+        );
+        elevatorSimulation.setInputVoltage(elevatorMotorVoltage);
+    }
```

Comment:
Don't use the power to voltage function, just make the calculations return voltage (and rename things in this function relative to that change)